File: backend/app/models/platform_account.rb
# == Schema Information
#
# Table name: platform_accounts
#
#  id          :bigint           not null, primary key
#  nickname    :string(100)      not null
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#  account_id  :string(50)
#  platform_id :bigint           not null
#  user_id     :bigint           not null
#
# Indexes
#
#  index_platform_accounts_on_platform_id  (platform_id)
#  index_platform_accounts_on_user_id      (user_id)
#
# Foreign Keys
#
#  fk_rails_...  (platform_id => platforms.id) ON DELETE => restrict
#  fk_rails_...  (user_id => users.id) ON DELETE => cascade
#
class PlatformAccount < ApplicationRecord
  belongs_to :user
  belongs_to :platform

  has_many :investments, dependent: :nullify

  validates :nickname, presence: true
end

File: backend_python/app/models/platform_account.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class PlatformAccount(Base):
    """A user's account on an investment platform — user ↔ platform junction."""

    __tablename__ = "platform_accounts"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(
        ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False
    )
    platform_id: Mapped[int] = mapped_column(
        ForeignKey("platforms.id", ondelete="RESTRICT"), nullable=False
    )
    nickname: Mapped[str] = mapped_column(String(100), nullable=False)
    account_id: Mapped[str | None] = mapped_column(String(50), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

    platform = relationship("Platform", back_populates="platform_accounts")
    user = relationship("User", back_populates="platform_accounts")
    investments = relationship("Investment", back_populates="platform_account")
